toolchain/check/eval.cpp

```diff
@@ -2066,16 +2066,22 @@ auto TryEvalTypedInst<SemIR::WhereExpr>(EvalContext& eval_context,
       } else if (auto impls =
                      eval_context.insts().TryGetAs<SemIR::RequirementImpls>(
                          inst_id)) {
-        if (IsPeriodSelf(eval_context,
-                         eval_context.constant_values().Get(impls->lhs_id))) {
-          if (impls->rhs_id == SemIR::TypeType::TypeInstId) {
+        SemIR::ConstantId lhs_const_id =
+            eval_context.GetConstantValue(impls->lhs_id);
+        SemIR::ConstantId rhs_const_id =
+            eval_context.GetConstantValue(impls->rhs_id);
+        if (IsPeriodSelf(eval_context, lhs_const_id)) {
+          auto rhs_inst_id =
+              eval_context.constant_values().GetInstId(rhs_const_id);
+          if (rhs_inst_id == SemIR::ErrorInst::InstId) {
+            // `.Self impls <error>` -> nothing to do.
```

Comment:
If we drop this entirely, doesn't that mean that `I where .Self impls <error>` will constant-evaluate to `I` rather than to` <error>`? Maybe we should set `phase` to `Error` here, or directly return `ErrorInst::ConstantId`?

Reply:
Oh good point. Rather than a whereexpr with an error in it, we want the constant value to be an error. Done. Added a test that shows it too.